lib/charms/opensearch/v0/helper_cluster.py

```diff
@@ -155,8 +155,18 @@ def nodes(
         opensearch: OpenSearchDistribution,
         use_localhost: bool,
         hosts: Optional[List[str]] = None,
+        only_this_juju_app: str = None,
```

Comment:
i believe the caller should be the one filtering the nodes belonging to a cluster if needed. Which should be pretty simple: 
`[node for node in nodes() if node.app_name == APP_NAME]`